src/shared/Core/Authentication/OAuth/OAuth2Client.cs

```diff
@@ -332,7 +350,7 @@ protected virtual bool TryCreateExceptionFromResponse(string json, out OAuth2Exc
 
         #region Helpers
 
-        private HttpRequestMessage CreateRequestMessage(HttpMethod method, Uri requestUri, HttpContent content = null, bool addAuthHeader = false)
+        protected HttpRequestMessage CreateRequestMessage(HttpMethod method, Uri requestUri, HttpContent content = null, bool addAuthHeader = false)

```

Comment:
Do we still need this to be protected if we can disable auth headers on token requests via the ctor?

Reply:
No, good spot.